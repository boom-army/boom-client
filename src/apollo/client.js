import { ApolloClient, InMemoryCache } from "@apollo/client";
import { offsetLimitPagination } from "@apollo/client/utilities";

export const client = new ApolloClient({
  cache: new InMemoryCache({
    typePolicies: {
      Query: {
        fields: {
          feed: offsetLimitPagination(),
          users: offsetLimitPagination(),
        },
      },
    },
  }),
  uri: process.env.REACT_APP_APOLLO_API || "http://locahost:7777",
  headers: {
    authorization: localStorage.getItem("token") || "",
  },
  context: async ({ req }) => {
    const token = req?.headers?.authorization || null;
    const user= token ? JSON.parse(localStorage.getItem("user")) : null;
    return { user, isLoggedIn: !!token };
  },
});
