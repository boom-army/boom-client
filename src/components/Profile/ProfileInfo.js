import React from "react";
import { Link } from "react-router-dom";
import styled from "styled-components";
import CoverPhoto from "../../styles/CoverPhoto";
import UserAvatar from "../UserAvatar";
import { Button, Box, Grid } from "@mui/material";
import { Follow } from "./Follow";
import { DobIcon, LocationIcon, LinkIcon } from "../Icons";

const Wrapper = styled.div`
  border-bottom: 1px solid ${(props) => props.theme.tertiaryColor};
  position: relative;

  .row {
    padding-bottom: 1rem;
  }

  .avatar {
    margin-left: 1.4rem;
    margin-top: -4rem;
  }

  .profile-name-handle {
    margin-left: 1.4rem;

    span.consumerName {
      font-weight: bold;
    }

    span.wallet {
      margin-top: 0.1rem;
      color: ${(props) => props.theme.secondaryColor};
    }

    span.handle {
      margin-top: 0.1rem;
      color: ${(props) => props.theme.secondaryColor};
    }
  }

  .profile-info {
    padding: 0 1.4rem 1rem;

    .bio {
      width: 90%;
    }
  }

  .action-btn {
    position: absolute;
    right: 1rem;
  }

  div.loc-dob-web {
    display: flex;
    color: ${(props) => props.theme.secondaryColor};
    margin: 0.6rem 0;

    span {
      margin-right: 1.5rem;
    }

    svg {
      margin-right: 0.2rem;
      position: relative;
      top: 3px;
    }
  }

  div.follow-following {
    color: ${(props) => props.theme.secondaryColor};
    span {
      margin-right: 1.3rem;
    }
  }

  @media screen and (max-width: 530px) {
    div.loc-dob-web {
      display: flex;
      flex-direction: column;

      span {
        margin-bottom: 0.7rem;
      }
    }
  }
`;

const ProfileInfo = ({ profile }) => {
  const {
    id,
    coverPhoto,
    avatar,
    bio,
    location,
    website,
    isSelf,
    dob,
    isFollowing,
    followersCount,
    followingCount,
    handle,
    consumerName,
    publicAddress,
  } = profile;

  return (
    <Wrapper>
      <Grid
        container
        direction="row"
        className="row"
        justifyContent="space-between"
        alignItems="flex-start"
      >
        <CoverPhoto src={coverPhoto} alt="cover" />
        <UserAvatar avatar={avatar} />
      </Grid>

      <Grid container direction="row" className="row" alignItems="flex-start">
        {isSelf ? (
          <Box pl={3}>
            <Button
              component={Link}
              to="/settings/profile"
              variant="outlined"
              color="primary"
              size="small"
            >
              Edit Profile
            </Button>
          </Box>
        ) : (
          <Box p={2} pl={4}>
            <Follow isFollowing={isFollowing} id={id} />
          </Box>
        )}
      </Grid>
      <Grid container direction="row" className="row" alignItems="flex-start">
        <div className="profile-name-handle">
          <span className="consumerName">{consumerName}</span>
          <br />
          <span className="wallet">
            <strong>wallet: </strong>
            {publicAddress}
          </span>
          <br />
          <span className="handle">{`@${handle}`}</span>
        </div>
      </Grid>
      <Grid
        container
        direction="row"
        justifyContent="space-between"
        alignItems="flex-start"
        className="profile-info"
      >
        <p className="bio">{bio}</p>

        {!location && !website && !dob ? null : (
          <div className="loc-dob-web">
            {location ? (
              <span>
                <LocationIcon /> {location}
              </span>
            ) : null}

            {website ? (
              <span>
                <LinkIcon /> {website}
              </span>
            ) : null}

            {dob ? (
              <span>
                <DobIcon />
                {dob}
              </span>
            ) : null}
          </div>
        )}

        <div className="follow-following">
          <span>
            {followersCount ? `${followersCount} followers` : "No followers"}
          </span>
          <span>
            {followingCount
              ? `${followingCount} following`
              : "Not following anyone"}
          </span>
        </div>
      </Grid>
    </Wrapper>
  );
};

export default ProfileInfo;
