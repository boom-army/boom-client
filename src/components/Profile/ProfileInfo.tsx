import React from "react";
import { Link } from "react-router-dom";
import CoverPhoto from "../../styles/CoverPhoto";
import UserAvatar from "../UserAvatar";
import { Button, Box, Grid } from "@mui/material";
import { Follow } from "./Follow";
import { DobIcon, LocationIcon, LinkIcon } from "../Icons";
import { styled } from "@mui/material/styles";
import { relative } from "path/win32";

const Wrapper = styled("div")((props) => ({
  // borderBottom: `1px solid ${props.theme.tertiaryColor}`,

  ".avatar": {
    // marginLeft: "1.4rem",
    // marginTop: "-4rem",
  },

  ".profile-name-handle": {
    wordBreak: "break-all",

    "span.consumerName": {
      fontWeight: "bold",
    },

    "span.wallet": {
      marginTop: "0.1rem",
      color: props.theme.secondaryColor,
    },

    "span.handle": {
      marginTop: "0.1rem",
      color: props.theme.secondaryColor,
    },
  },

  ".profile-info": {
    padding: "0 1.4rem 1rem",

    ".bio": {
      width: "100%",
    },
  },

  "div.loc-dob-web": {
    display: "flex",
    color: props.theme.secondaryColor,
    margin: "0.6rem 0",

    span: {
      marginRight: "1.5rem",
    },

    svg: {
      marginRight: "0.2rem",
      position: "relative",
      top: "3px",
    },
  },

  "div.follow-following": {
    color: props.theme.secondaryColor,
    span: {
      marginRight: "1.3rem",
      color: "#4d97cb",
      borderBottom: "1px solid #4d97cb",
      textTransform: "capitalize",
      padding: "0 0 3px 0",
    },
  },
  ".wallet-topbar": {
    backgroundColor: "#4d97cb",
    color: "#0e1b25",
    padding: "7px",
    fontSize: "13px",
    ".wallet": {
      display: "flex",
      justifyContent: "space-between",
      svg: {
        verticalAlign: "middle",
        marginRight: "5px",
      },
      ".copyIcon": {
        svg: {
          marginLeft: "10px",
        },
      },
    },
  },
  ".userimage-section": {
    display: "flex",
    alignItems: "center",
    gap: "10px",
    marginLeft: "1.4rem",
  },
  ".profile-Button": {
    borderRadius: "25px",
    padding: "4px 5px",
    fontSize: "12px",
    color: "#4d97cb",
    borderColor: "#4d97cb",
    textTransform: "capitalize",
  },
  ".userImg": {
    position: "relative",
    ".userChipBdr": {
      backgroundColor: "#315981",
      display: "flex",
      flexShrink: "0",
      height: "calc(40px + 8px)",
      width: "calc(40px + 8px)",
      clipPath:
        "polygon(50% 0%, 83% 12%, 100% 43%, 94% 78%, 68% 100%, 32% 100%, 6% 78%, 0% 43%, 17% 12%)",
    },
    ".MuiAvatar-root": {
      width: "calc(25px + 19px)",
      height: "calc(25px + 19px)",
      position: "relative",
      top: "2px",
      left: "2px",
      clipPath:
        "polygon(50% 0%, 83% 12%, 100% 43%, 94% 78%, 68% 100%, 32% 100%, 6% 78%, 0% 43%, 17% 12%);",
      borderRadius: "0px",
      img: {
        width: "100%",
        height: "100%",
        objectFit: "cover",
      },
    },
    ".tokenImg": {
      img: {
        width: "20px",
        height: "20px",
        borderRadius: "50%",
        position: "absolute",
        right: "0",
        bottom: "-6px",
        border: "3px solid #15202b",
      },
    },
  },
  ".bio": {
    color: props.theme.secondaryColor,
  },

  "@media screen and (max-width: 767px)": {
    ".wallet-topbar": {
      whiteSpace: "nowrap",
      ".wallet": {
        fontSize: "10px",
      },
    },
  },
}));

const ProfileInfo = ({ profile }: any) => {
  const {
    id,
    coverPhoto,
    avatar,
    bio,
    location,
    website,
    isSelf,
    dob,
    isFollowing,
    followersCount,
    followingCount,
    handle,
    consumerName,
    publicAddress,
  } = profile;

  return (
    <Wrapper>
      <Grid container>
        <Grid item xs={12}>
          <div className="wallet-topbar">
            <span className="wallet">
              <span>
                <svg
                  xmlns="http://www.w3.org/2000/svg"
                  aria-hidden="true"
                  role="img"
                  width="18px"
                  height="18px"
                  preserveAspectRatio="xMidYMid meet"
                  viewBox="0 0 256 256"
                >
                  <path
                    fill="currentColor"
                    d="M164 144a16 16 0 1 1 16 16a16 16 0 0 1-16-16Zm72-48v104a20.1 20.1 0 0 1-20 20H56a28.1 28.1 0 0 1-28-28V68.2A32.1 32.1 0 0 1 60 36h132a12 12 0 0 1 0 24H60a8.4 8.4 0 0 0-5.8 2.4A8.2 8.2 0 0 0 52 68v.2a8.4 8.4 0 0 0 8.5 7.8H216a20.1 20.1 0 0 1 20 20Zm-24 4H60.5a33.5 33.5 0 0 1-8.5-1.1V192a4 4 0 0 0 4 4h156Z"
                  />
                </svg>

                {publicAddress}
              </span>

              <span className="copyIcon">
                <svg
                  xmlns="http://www.w3.org/2000/svg"
                  aria-hidden="true"
                  role="img"
                  width="15px"
                  height="15px"
                  preserveAspectRatio="xMidYMid meet"
                  viewBox="0 0 24 24"
                >
                  <path
                    fill="#316081"
                    d="M7 6V3a1 1 0 0 1 1-1h12a1 1 0 0 1 1 1v14a1 1 0 0 1-1 1h-3v3c0 .552-.45 1-1.007 1H4.007A1.001 1.001 0 0 1 3 21l.003-14c0-.552.45-1 1.007-1H7zM5.003 8L5 20h10V8H5.003zM9 6h8v10h2V4H9v2z"
                  />
                </svg>
              </span>
            </span>
          </div>
        </Grid>
      </Grid>
      <Grid container pt={2}>
        <Grid item xs={8}>
          <div className="userimage-section">
            <span className="userImg">
              <span className="userChipBdr">
                <UserAvatar avatar={avatar} />
              </span>
              <span className="tokenImg">
                <img alt="cover" src="https://ec4meandtybfo3zpqizry3nes5efdzazxhucv6jrjvc5n4igcy.arweave.net/ILjCAaOeAldvL4IzHG2k_l0hR5Bm56Cr5MU1F1vEGFo?ext=png" />
              </span>
            </span>
            <div className="profile-name-handle">
              <span className="consumerName">{consumerName}</span>
              <br />
              <span className="handle">{`@${handle}`}</span>
            </div>
          </div>
        </Grid>

        <Grid item xs={4} pr={3} sx={{ textAlign: "right" }}>
          {isSelf ? (
            <Button
              className="profile-Button"
              component={Link}
              to="/settings/profile"
              variant="outlined"
              color="primary"
              size="small"
            >
              Edit Profile
            </Button>
          ) : (
            <Follow isFollowing={isFollowing} id={id} />
          )}
        </Grid>
      </Grid>
      {/* <Box mb={3}>
        <CoverPhoto src={coverPhoto} alt="cover" />
        <UserAvatar avatar={avatar} />
      </Box> */}
      <Box pl={3}>
        {/* <Box>
          {isSelf ? (
            <Button
              component={Link}
              to="/settings/profile"
              variant="outlined"
              color="primary"
              size="small"
            >
              Edit Profile
            </Button>
          ) : (
            <Follow isFollowing={isFollowing} id={id} />
          )}
        </Box> */}
        <Box mt={1} display={"inline-flex"}>
          <div className="profile-name-handle">
            {/* <span className="consumerName">{consumerName}</span>
            <br /> */}
            {/* <span className="wallet">
              <strong>wallet: </strong>
              {publicAddress}
            </span>
            <br /> */}
            {/* <span className="handle">{`@${handle}`}</span> */}
          </div>
        </Box>
        <Box pb={1}>
          <p className="bio">{bio}</p>

          {!location && !website && !dob ? null : (
            <div className="loc-dob-web">
              {location ? (
                <span>
                  <LocationIcon /> {location}
                </span>
              ) : null}

              {website ? (
                <span>
                  <LinkIcon /> {website}
                </span>
              ) : null}

              {dob ? (
                <span>
                  <DobIcon />
                  {dob}
                </span>
              ) : null}
            </div>
          )}

          <div className="follow-following">
            <span>
              {followersCount ? `${followersCount} followers` : "No followers"}
            </span>
            <span>
              {followingCount
                ? `${followingCount} following`
                : "Not following anyone"}
            </span>
          </div>
        </Box>
      </Box>
    </Wrapper>
  );
};

export default ProfileInfo;
