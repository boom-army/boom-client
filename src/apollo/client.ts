import { ApolloClient, InMemoryCache } from "@apollo/client";
import { offsetLimitPagination } from "@apollo/client/utilities";

export const client = new ApolloClient({
  cache: new InMemoryCache({
    typePolicies: {
      Query: {
        fields: {
          feed: offsetLimitPagination(),
          heroFeed: offsetLimitPagination(),
          getChannelById: offsetLimitPagination(),
          users: offsetLimitPagination(),
          searchTweets: offsetLimitPagination(),
          searchUser: offsetLimitPagination(),
          mentions: offsetLimitPagination(),
        },
      },
    },
  }),
  credentials: 'include',
  uri: process.env.REACT_APP_APOLLO_API || "http://locahost:7777",
  headers: {
    authorization: localStorage.getItem("token") || "",
    "Access-Control-Allow-Origin" : "*",
  },
  defaultOptions: {
    watchQuery: {
      context: async ({ req }: any) => {
        const token = req?.headers?.authorization || null;
        const user = token ? localStorage.getItem("user")?.toString() : null;
        return { user, isLoggedIn: !!token };
      },
    },
  },
});
