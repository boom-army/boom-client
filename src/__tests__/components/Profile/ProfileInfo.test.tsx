import React from "react";
import ProfileInfo from "../../../components/Profile/ProfileInfo";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import { MockedProvider } from "@apollo/client/testing";
import { act, fireEvent, render, screen } from "@testing-library/react";

const profile: any = {
  data: {
    profile: {
      id: "ckzny1iv10031lmn568gj3b3n",
      publicAddress: "JBZ52cKhHiFJdzQBNWnp3Xy2jHgDgkhWwSLWoNgLNxD4",
      handle: "shy-cloud-4965",
      consumerName: "shy-cloud-4965",
      avatar: "",
      coverPhoto: "/default-cover.png",
      dob: "",
      location: "",
      website: "",
      isSelf: true,
      isFollowing: false,
      followersCount: 0,
      followingCount: 0,
      tweetsCount: 21,
      newMentionsCount: 0,
      bio: "full stack developer",
      tweets: [
        {
          id: "cl0c77yow71753s0n5h032nerw",
          text: "good scope.",
          tags: [],
          isTweetMine: true,
          user: {
            id: "ckzny1iv10031lmn568gj3b3n",
            consumerName: "shy-cloud-4965",
            publicAddress: "JBZ52cKhHiFJdzQBNWnp3Xy2jHgDgkhWwSLWoNgLNxD4",
            handle: "shy-cloud-4965",
            avatar: "",
            __typename: "User",
          },
          files: [],
          gif: null,
          reactions: [],
          commentsCount: 0,
          retweetsCount: 0,
          isRetweet: false,
          tipsCount: "0",
          createdAt: "1646385168992",
          __typename: "Tweet",
        },
        {
          id: "cl084xdy815761s0n5omoxoem0",
          text: "testing after changes in input converted to ts",
          tags: [],
          isTweetMine: true,
          user: {
            id: "ckzny1iv10031lmn568gj3b3n",
            consumerName: "shy-cloud-4965",
            publicAddress: "JBZ52cKhHiFJdzQBNWnp3Xy2jHgDgkhWwSLWoNgLNxD4",
            handle: "shy-cloud-4965",
            avatar: "",
            __typename: "User",
          },
          files: [],
          gif: null,
          reactions: [
            {
              id: "cl0c774p371312s0n533u0b29k",
              emojiId: "+1",
              skin: 1,
              isMine: true,
              count: 1,
              __typename: "Reaction",
            },
          ],
          commentsCount: 1,
          retweetsCount: 0,
          isRetweet: false,
          tipsCount: "0",
          createdAt: "1646139451616",
          __typename: "Tweet",
        },
        {
          id: "cl081ymfz8867s0n59y782hdj",
          text: "🥰",
          tags: [],
          isTweetMine: true,
          user: {
            id: "ckzny1iv10031lmn568gj3b3n",
            consumerName: "shy-cloud-4965",
            publicAddress: "JBZ52cKhHiFJdzQBNWnp3Xy2jHgDgkhWwSLWoNgLNxD4",
            handle: "shy-cloud-4965",
            avatar: "",
            __typename: "User",
          },
          files: [],
          gif: null,
          reactions: [],
          commentsCount: 0,
          retweetsCount: 0,
          isRetweet: false,
          tipsCount: "0",
          createdAt: "1646134470431",
          __typename: "Tweet",
        },
        {
          id: "cl06ku5qr9948ban5356ugu9b",
          text: "😍",
          tags: [],
          isTweetMine: true,
          user: {
            id: "ckzny1iv10031lmn568gj3b3n",
            consumerName: "shy-cloud-4965",
            publicAddress: "JBZ52cKhHiFJdzQBNWnp3Xy2jHgDgkhWwSLWoNgLNxD4",
            handle: "shy-cloud-4965",
            avatar: "",
            __typename: "User",
          },
          files: [],
          gif: null,
          reactions: [],
          commentsCount: 0,
          retweetsCount: 0,
          isRetweet: false,
          tipsCount: "0",
          createdAt: "1646045242515",
          __typename: "Tweet",
        },
        {
          id: "ckzzg3lm6169405vun5im00fsmq",
          text: "😘6210",
          tags: [],
          isTweetMine: true,
          user: {
            id: "ckzny1iv10031lmn568gj3b3n",
            consumerName: "shy-cloud-4965",
            publicAddress: "JBZ52cKhHiFJdzQBNWnp3Xy2jHgDgkhWwSLWoNgLNxD4",
            handle: "shy-cloud-4965",
            avatar: "",
            __typename: "User",
          },
          files: [],
          gif: null,
          reactions: [],
          commentsCount: 0,
          retweetsCount: 0,
          isRetweet: false,
          tipsCount: "0",
          createdAt: "1645614021678",
          __typename: "Tweet",
        },
      ],
      createdAt: "1644918563773",
      __typename: "User",
    },
  },
};

describe("user profile info", () => {
  test("renders correctly user", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <ProfileInfo profile={profile.data.profile} />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

test("Assertion testing of <ProfileInfo/> component", () => {
  render(<BrowserRouter>
    <ProfileInfo profile={profile.data.profile} />
  </BrowserRouter>);

  // screen.getByRole('');
  expect(screen.getByRole('img', { name: 'cover' })).toHaveAttribute('src', '/default-cover.png');
  expect(screen.getByRole('link', { name: 'Edit Profile' })).toHaveAttribute('href', '/settings/profile');

});
