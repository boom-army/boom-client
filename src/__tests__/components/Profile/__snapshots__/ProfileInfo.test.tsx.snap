// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user profile info renders correctly user 1`] = `
<div
  className="css-1jg57c8"
>
  <div
    className="MuiBox-root css-i3pbo"
  >
    <img
      alt="cover"
      className="css-3y6f34"
      src="/default-cover.png"
    />
    <div
      className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault avatar css-2s90m6-MuiAvatar-root"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="PersonIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
        />
      </svg>
    </div>
  </div>
  <div
    className="MuiBox-root css-1ns5f5t"
  >
    <div
      className="MuiBox-root css-0"
    >
      <a
        className="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonBase-root  css-1k23hlb-MuiButtonBase-root-MuiButton-root"
        href="/settings/profile"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
      >
        Edit Profile
      </a>
    </div>
    <div
      className="MuiBox-root css-133i6oc"
    >
      <div
        className="profile-name-handle"
      >
        <span
          className="consumerName"
        >
          shy-cloud-4965
        </span>
        <br />
        <span
          className="wallet"
        >
          <strong>
            wallet: 
          </strong>
          JBZ52cKhHiFJdzQBNWnp3Xy2jHgDgkhWwSLWoNgLNxD4
        </span>
        <br />
        <span
          className="handle"
        >
          @shy-cloud-4965
        </span>
      </div>
    </div>
    <div
      className="MuiBox-root css-4cxybv"
    >
      <p
        className="bio"
      >
        full stack developer
      </p>
      <div
        className="follow-following"
      >
        <span>
          No followers
        </span>
        <span>
          Not following anyone
        </span>
      </div>
    </div>
  </div>
</div>
`;
